packages/kbn-config-schema/src/types/type.ts

```diff
@@ -6,15 +6,35 @@
  * Side Public License, v 1.
  */
 
+import type { AnySchema, CustomValidator, ErrorReport } from 'joi';
 import { SchemaTypeError, ValidationError } from '../errors';
-import { AnySchema, internals, ValidationErrorItem } from '../internals';
+// import { internals } from '../internals';
 import { Reference } from '../references';
 
 export interface TypeOptions<T> {
   defaultValue?: T | Reference<T> | (() => T);
   validate?: (value: T) => string | void;
 }
 
+export const convertValidationFunction = <T = unknown>(
+  validate: (value: T) => string | void
+): CustomValidator<T> => {
+  return (value, { error }) => {
+    let validationResultMessage;
+    try {
+      validationResultMessage = validate(value);
```

Comment:
We were overriding the base `any.custom` rule in our `any` type to change its signature, which is no longer allowed (you can't override a rule from the root type in custom types), so  we're converting before callings `schema.custom`